app/ydoc-shared/src/ast/parse.ts

```diff
@@ -705,6 +705,7 @@ export function repair(
   const fixes = module ?? root.module.edit()
   for (const ast of lostInline) {
     if (ast instanceof Group) continue
+    if (ast instanceof BodyBlock) continue
```

Comment:
Ended up moving the condition to `checkSpans`, as I wanted to avoid filtering the whole nested map of sets ahead of time. Since we are already looping through "lost" nodes, it is much cheaper to exclude them during that loop. I think that still matches your intention of having this "valid mismatches" logic in one place and it never hitting repair in the first place.